### PR TITLE
Switch to Vite PWA Nuxt plugin

### PR DESCRIPTION
**Pull request by Siilwyn**
https://trello.com/c/B84BvDTQ/10-revise-pwa-plugin

Moved google HTML file to meta tag since otherwise the PWA plugin complains about it not being precached, which we don't want. Could've put it in an ignore config but figured meta is the way to go.

Also added a SVG favicon and apple touch icon for good measure.

### CODE REVIEW

**Review by kyrel — COMMENTED** [review]
The Nuxt 2 version had this thing which I could not replicate with the Nuxt 3 plugin I've used:

 ```
   workbox: {
      runtimeCaching: [
        {
          urlPattern: 'https://www.datocms-assets.com/.*',
          handler: 'cacheFirst',
          method: 'GET',
        },
      ],
    }
```
Is it doable with the plugin you're using?

**Comment by Siilwyn** [comment]
@kyrel yes it's possible! Added that back.

File: src/layouts/default.vue [hunk 1]
```diff
@@ -61,6 +62,16 @@ onBeforeUnmount(() => {
...
+  link: [
+    { rel: 'icon', href: '/favicon.svg', type: 'image/svg+xml', },
+    { rel: 'apple-touch-icon', href: '/apple-touch-icon.png', },
+  ],
+  meta: [
+    {
+      name: "google-site-verification",
+      content: "zpj1hb-uMNGkFLJlxqC96oWE067G1lzXcRFsWM4m_8M",
+    },
+  ],
```
Comment (kyrel) [inline src/layouts/default.vue:74]: I wonder if there's a reason to put it here and not in nuxt config (have the same question about the script below as well in fact :) )

File: src/layouts/default.vue [hunk 2]
```diff
@@ -61,6 +62,16 @@ onBeforeUnmount(() => {
 });
 
 useHead({
+  link: [
+    { rel: 'icon', href: '/favicon.svg', type: 'image/svg+xml', },
+    { rel: 'apple-touch-icon', href: '/apple-touch-icon.png', },
```
[the remark this answers is not in the document]
Comment (Siilwyn) [reply src/layouts/default.vue:67]: > this one does not autogenerate the icon link tags
yes, that's why we now do this ourselves

**Review by jbmoelker — APPROVED** [review]

**Pull request merged**